### ISSUE TITLE
Error with UglifyJs

### ISSUE THREAD

**Issue by Mango3403** [cssivision/qrcode-react#13]
when I use webpack + UglifyJs to build. something error

**ERROR in js/main-2a80142.js from UglifyJs
Unexpected token: name (QRCode) [js/main-2a80142.js:92156,6]**

**log**
```
PS C:\custom> npm run build

custom-webpack3@1.0.0 build C:\custom
webpack --progress --profile --colors --config webpack.pro.config.js

clean-webpack-plugin: C:\custom\public\pro has been removed.
10329ms building modules
20ms sealing
1ms optimizing
0ms basic module optimization
15ms module optimization
1ms advanced module optimization
0ms basic chunk optimization
1ms chunk optimization
6ms advanced chunk optimization
89ms building modules
21ms module and chunk tree optimization
1ms chunk modules optimization
0ms advanced chunk modules optimization
13ms module reviving
2ms module order optimization
11ms module id optimization
0ms chunk reviving
1ms chunk order optimization
19ms chunk id optimization
59ms hashing
1ms module assets processing
307ms chunk assets processing
2ms additional chunk assets processing
1ms recording
1138ms additional asset processing
1ms chunk asset optimization
334ms asset optimization
58ms emitting
Hash: 2a80142447f415564ba2
Version: webpack 3.6.0
Time: 12528ms
                                        Asset       Size  Chunks                    Chunk Names
      imagessrc/images/control/handle_del.png    3.21 kB          [emitted]
                      imagessrc/images/0.jpeg    94.3 kB          [emitted]
                      imagessrc/images/1.jpeg    54.8 kB          [emitted]
                      imagessrc/images/2.jpeg    59.1 kB          [emitted]
                      imagessrc/images/3.jpeg      36 kB          [emitted]
           imagessrc/images/control/clear.png    6.44 kB          [emitted]
              imagessrc/images/material/a.jpg    84.3 kB          [emitted]
           imagessrc/images/control/order.png     5.8 kB          [emitted]
                  imagessrc/images/upload.png    4.62 kB          [emitted]
   imagessrc/images/control/handle_rotate.png    3.59 kB          [emitted]
     imagessrc/images/control/handle_zoom.png    2.67 kB          [emitted]
   imagessrc/images/control/handle_change.png    17.7 kB          [emitted]
                           js/main-2a80142.js    2.86 MB       0  [emitted]  [big]  main
css/main.c1215285413f6483d42f9aae01fc3e97.css       1 kB       0  [emitted]         main
                                   index.html  702 bytes          [emitted]
                         assets/ruler.min.css    1.65 kB          [emitted]
                          assets/ruler.min.js    9.56 kB          [emitted]
  [29] (webpack)/buildin/global.js 509 bytes {0} [built]
       [] -> factory:0ms building:3ms = 3ms
 [347] ./src/images/0.jpeg 69 bytes {0} [built]
       [] -> factory:3307ms building:96ms = 3403ms
 [361] ./src/index.js + 45 modules 162 kB {0} [built]
 [459] ./src/style.css 133 bytes {0} [built]
       [] -> factory:609ms building:16ms = 625ms
 [737] ./src/images/1.jpeg 69 bytes {0} [built]
       [] -> factory:947ms building:43ms = 990ms
 [738] ./src/images/2.jpeg 69 bytes {0} [built]
       [] -> factory:946ms building:41ms = 987ms
 [739] ./src/images/3.jpeg 69 bytes {0} [built]
       [] -> factory:946ms building:40ms = 986ms
 [752] ./src/images/control/clear.png 80 bytes {0} [built]
       [] -> factory:3728ms building:104ms = 3832ms
 [753] ./src/images/material/a.jpg 77 bytes {0} [built]
       [] -> factory:776ms building:48ms = 824ms
 [757] ./src/images/control/order.png 80 bytes {0} [built]
       [] -> factory:458ms building:43ms = 501ms
 [758] ./src/images/control/handle_del.png 85 bytes {0} [built]
       [] -> factory:3568ms building:95ms = 3663ms
 [759] ./src/images/control/handle_rotate.png 88 bytes {0} [built]
       [] -> factory:3569ms building:95ms = 3664ms
 [760] ./src/images/control/handle_zoom.png 86 bytes {0} [built]
       [] -> factory:3569ms building:95ms = 3664ms
 [761] ./src/images/control/handle_change.png 88 bytes {0} [built]
       [] -> factory:3570ms building:94ms = 3664ms
 [816] ./node_modules/css-loader?{"modules":true}!./node_modules/postcss-loader/lib?{}!./src/style.css 1.26 kB [built]
       [] -> factory:2ms building:3217ms = 3219ms
    + 805 hidden modules

ERROR in js/main-2a80142.js from UglifyJs
Unexpected token: name (QRCode) [js/main-2a80142.js:92156,6]
Child html-webpack-plugin for "index.html":
     1 asset
       [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html 951 bytes {0} [built]
            factory:33ms building:34ms = 67ms
       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
           [] -> factory:629ms building:9ms = 638ms
       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
           [] -> factory:629ms building:7ms = 636ms
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--0-2
!node_modules/postcss-loader/lib/index.js??ref--0-3!src/style.css:
       [0] ./node_modules/css-loader?{"modules":true}!./node_modules/postcss-loader/lib?{"plugins":[null,{"version":"5.2.1
7","plugins":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,nul
l,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"postcssPlugin":"cssnano","postcssVersion":"5.2.1
7"}]}!./src/style.css 1.26 kB {0} [built]
            factory:7ms building:58ms = 65ms
        + 1 hidden module

npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run"
"build"
npm ERR! node v6.11.0
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! custom-webpack3@1.0.0 build: `webpack --progress --profile --colors --config webpack.pro.config.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the custom-webpack3@1.0.0 build script 'webpack --progress --profile --colors --config webpack.pro.conf
ig.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the custom-webpack3 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     webpack --progress --profile --colors --config webpack.pro.config.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs custom-webpack3
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls custom-webpack3
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\custom\npm-debug.log
```

**error file**
_line 92156_ 
```
class QRCode extends React.Component {
    shouldComponentUpdate(nextProps) {
        var that = this;
        return Object.keys(QRCode.propTypes).some(function(k) {
            return that.props[k] !== nextProps[k];
        });
    }

    componentDidMount() {
        this.update();
    }

...
```

**Comment by PrOmPt0** [comment]
`class QRCode extends React.Component {
    shouldComponentUpdate(nextProps) {
        var that = this;
        return Object.keys(QRCode.propTypes).some(function(k) {
            return that.props[k] !== nextProps[k];
        });
    }

    componentDidMount() {
        this.update();
    }`
it's ES6 
you should use babel-loader compile to ES5

**Comment by Mango3403** [comment]
ok, I fixed it. thank you. 发自网易邮箱大师 在2017年11月08日 09:34，PrOmPt0 写道： `class QRCode extends React.Component { shouldComponentUpdate(nextProps) { var that = this; return Object.keys(QRCode.propTypes).some(function(k) { return that.props[k] !== nextProps[k]; }); } componentDidMount() {
    this.update();
}`
 it's ES6 you should use babel-loader compile to ES5 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by richardriman** [comment]
I run into this issue too, no idea how to fix this. I already use babel-loader, in development where UglifyJS isn't in game, everything is OK. I use hundreds of npm packages and only this one make troubles to UglifyJS.

**Comment by cssivision** [comment]
fixed in 0.1.6
